### ISSUE TITLE
pygame 2.0 manylinux wheels are built without PulseAudio support

### ISSUE THREAD

**Issue by larryhastings** [pygame/pygame#1351]
As far as I can determine, the current pygame 2 wheel for manylinux x86_64 (pygame-2.0.0.dev3-cp37-cp37m-manylinux1_x86_64.whl) was built without PulseAudio support.  Why?

I'm using pygame 2.0.0.dev3 on an Ubuntu derivative called Pop!_OS, version 19.04, x86-64.

I wrote a C program that lists the audio drivers available in SDL2, and it shows me several, including "pulseaudio".  "pulseaudio" is the "current" audio driver.  (I built it using the SDL2 dev package available on my OS, version "2.0.9+dfsg1-1ubuntu1".)

I wrote a second C program that lists the audio devices available in SDL2.  By default, the audio devices it shows me roughly map to the audio devices I see in PulseAudio (e.g. on the "Sound" control panel, in the output of "pacmd list-sinks").  I can also force it to use the ALSA driver ("SDL_AUDIODRIVER=alsa") and then it shows me a different list of names.

I wrote a Python program that lists the audio devices using pygame 2.0.0.dev3, sneakily using "import pygame._sdl2".  By default, it shows me the list of audio devices I see when I force the second C program above to use the ALSA driver.  And if I run this Python program and force it to use the PulseAudio driver ("SDL_AUDIODRIVER=pulseaudio") pygame throws an exception: `sdl2.error: Audio system not initialised`.

(These C and Python programs are attached in the zip file.)

I cracked open the wheel listed above, and grepped for "pulseaudio", and didn't find any hits.  I recompiled my first C program above--the one that lists audio drivers--and linked it against the `libSDL2-2-ac339521.0.so.0.9.0` lurking in the wheel; the resulting program gives the "current" audio device as "alsa", and it _doesn't_ list the "pulseaudio" driver at all.

What's extra-weird is that the build process seems to install the pulseaudio development package in the Docker build environment.  In `buildconfig/manylinux-build/docker_base/Dockerfile-x86_64` I see it uses yum to install the `pulseaudio-libs-devel` package.  And the `--enable-pulseaudio` configure flag is on by default.  So it seems like it _should_ have been built with PulseAudio support.  But I can't find the build log for the manylinux wheels, nor the configure output, so I can't (currently) read for myself what's going on.

So... what's going on?  Is PulseAudio support disabled on purpose right now?  If so, will it return someday?

Thanks for your time!

[pygame2.manylinux.pulseaudio.zip](https://github.com/pygame/pygame/files/3666548/pygame2.manylinux.pulseaudio.zip)

**Comment by illume** [comment]
Hi. Thanks for writing up the issue  :) pulseaudio not working is considered a bug. 

It's possible we might need to build a version from source rather than rely on the one installed via YUM.

You can make pip show everything that's going on with a verbose flag. It's hiding all the build commands by default it seems. Inside buildconfig/manylinux-build/build-wheels.sh change this: `${PYBIN}/pip wheel -vvv /io/ -w wheelhouse/`

Anyone else on a Linux able to confirm if it's working/broken?

**Comment by illume** [comment]
Oops. A comment in a PR closed this by accident.

**Comment by illume** [comment]
I did some testing and tried the new manylinux2010 based builds, and those didn't magically fix things. The pygame 1.9.6 wheels that use SDL1 also lack pulseaudio support.

**Comment by illume** [comment]
I merged a commit some time ago that `SDL_AUDIODRIVER=pulse` works as it did with SDL1 like `SDL_AUDIODRIVER=pulseaudio` with SDL2.

Then, I noticed that the library paths on ubuntu are perhaps different to what it expects. This is a work around that allowed midi to work via pulse:
`sudo ln -s /usr/lib/i386-linux-gnu/alsa-lib /usr/lib/alsa-lib`

So, the pulse audio library paths are something to research more.
